### PR TITLE
ECOSYS-77 Payara Server setup support in VSCode

### PR DESCRIPTION
**Pull request by jGauravGupta**
# Description
This is an initial feature to add/remove/rename the Payara Server to/from VS Code.

# Test

- Install Node and Yarn
- Open a terminal and run `yarn install`  to install all dependencies for the project.
- Run `code` command from the terminal to open the project in VS Code.
- Press `F5` from Code to run the project which will start another instance of Code with an installed plugin.
- Press `Ctrl + Shift + P` and search for `Add Payara Server` option.
image
- Browse the valid Payara path and enter the name.
image

### CODE REVIEW

**Comment by jGauravGupta** [comment]
@smillidge review needed for licensing.

File: src/main/fish/payara/server/PayaraInstanceController.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        return (input: ui.MultiStepInput) => this.selectDomain(input, state);
+    }
+
+    async selectDomain(input: ui.MultiStepInput, state: Partial<State>) {
+        const title = 'Register Payara Server';
+        const pick = await input.showQuickPick({
+            title,
+            step: 2,
+            totalSteps: 3,
+            placeholder: 'Select a exisitng domain',
```
Comment (MattGill98) [inline src/main/fish/payara/server/PayaraInstanceController.ts]: Just a minor typo

File: src/main/UI.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		}
+		if (this.current) {
+			this.current.dispose();
+		}
+	}
+
+	public async showQuickPick<T extends QuickPickItem, P extends QuickPickParameters<T>>({ title, step, totalSteps, items, activeItem, placeholder, buttons, shouldResume }: P) {
+		const disposables: Disposable[] = [];
+		try {
+			return await new Promise<T | (P extends { buttons: (infer I)[] } ? I : never)>((resolve, reject) => {
```
Comment (MattGill98) [inline src/main/UI.ts]: I think you can remove the `async` and `await` keywords here, they don't seem to be doing anything

File: src/main/UI.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+			});
+		} finally {
+			disposables.forEach(d => d.dispose());
+		}
+	}
+
+	public async showInputBox<P extends InputBoxParameters>({ title, step, totalSteps, value, prompt, validate, buttons, shouldResume }: P) {
+		const disposables: Disposable[] = [];
+		try {
+			return await new Promise<string | (P extends { buttons: (infer I)[] } ? I : never)>((resolve, reject) => {
```
Comment (MattGill98) [inline src/main/UI.ts]: Same here

File: package.json [hunk 4]
```diff
@@ -3,21 +3,110 @@
...
+			{
+				"command": "payara.server.remove.context",
+				"title": "Remove"
+			},
+			{
+				"command": "payara.server.rename.context",
+				"title": "Rename"
+			}
+		],
+		"views": {
```
Comment (MattGill98) [inline package.json]: I'm not personally a fan of this menu being in the explorer view (it's not related in any way to the open folders).

I'd personally favour this appearing in the `viewsContainers` contribution point, and then selecting and adding servers like how you manage debug configurations

File: src/main/UI.ts (hunk 2 above)
Comment (jGauravGupta) [reply src/main/UI.ts]: Cross verified, after removing these keywords, UI panel does not perform anymore.

File: src/main/UI.ts (hunk 3 above)
Comment (jGauravGupta) [reply src/main/UI.ts]: Same as above.

File: package.json (hunk 4 above)
Comment (jGauravGupta) [reply package.json]: Added the Payara Server activity bar to `viewsContainers` but I would also recommend keeping Payara Server in explorer too, near to opened applications.

**Pull request merged**